playbooks/roles/common/tasks/repositories.yml

```diff
@@ -1,9 +1,21 @@
 ---
 - name: 'Setup Epel Repository'
   yum: name=http://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ release }}.noarch.rpm state=present
-- name: 'Setup Puppetlabs Repository'
-  yum: name=http://yum.puppetlabs.com/puppetlabs-release-el-{{ release }}.noarch.rpm state=present
+
+- name: 'Setup Puppet 3 Repository'
+  yum:
+    name: http://yum.puppetlabs.com/puppetlabs-release-el-{{ release }}.noarch.rpm
+    state: present
+  when: common_puppet_version is undefined or common_puppet_version == 3
+
+- name: 'Setup Puppet 4 Repository'
+  yum:
+    name: https://yum.puppetlabs.com/puppetlabs-release-pc1-el-{{ release }}.noarch.rpm
+    state: present
+  when: common_puppet_version is defined and common_puppet_version == 4
```

Comment:
I'd prefer we make a `puppet_repositories` role that sets this up and handles it. I'd like to downplay and remove common all together given its lack of expressiveness. I don't mind re-factoring this myself after if you prefer.